src/riscv/lib/src/state_backend/proof_backend/proof/deserialise_owned.rs

```diff
@@ -103,12 +109,12 @@ impl ProofTreeDeserialiser<'_> {
 
 /// Suspended computation combinator for [`ProofTreeDeserialiser`] deserialiser.
 pub struct OwnedParserComb<'t, R> {
-    result: R,
+    result: ParseProofResult<R>,
```

Comment:
What's the motivation behind moving the fallibility into the struct?

Reply:
Otherwise, the separation between error types wouldn't be a separation, but rather a nested doll type of approach. Now, `DeserResult` & `ProofParseResult` have distinct error types with distinct variants (whereas otherwise, `DeserResult` would have a variant for `ProofParseResult`)

 I would say this is an improvement overall - having the minimal error type needed in each place & making both implementations align better with eachother.

Reply:
Do you think this may cause problems with suspended computations that can't fail? Do those even exist?